source/opt/aggressive_dead_code_elim_pass.cpp

```diff
@@ -513,6 +513,26 @@ bool AggressiveDCEPass::AggressiveDCE(Function* func) {
       AddBranch(mergeBlockId, *bi);
       for (++bi; (*bi)->id() != mergeBlockId; ++bi) {
       }
+
+      auto merge_terminator = (*bi)->terminator();
+      if (merge_terminator->opcode() == SpvOpUnreachable) {
+        // The merge was unreachable. This is undefined behaviour so just
+        // return (or return an undef). Then mark the new return as live.
+        auto func_ret_type_inst = get_def_use_mgr()->GetDef(func->type_id());
+        if (func_ret_type_inst->opcode() == SpvOpTypeVoid) {
+          merge_terminator->SetOpcode(SpvOpReturn);
+        } else {
+          // Find an undef for the return value and make sure it gets kept by
+          // the pass.
+          auto undef_id = Type2Undef(func->type_id());
```

Comment:
The original code is likely to be an error, if it gets executed.  I'm wondering if it might be better to return an OpConstantNull instead of an Undef.  It might make debugging easier if this is ever hit.

Reply:
Alan pointed out that not every type can be used in an OpConstantNull, so we should stick with the undef.